homeassistant/components/apple_tv/config_flow.py

```diff
@@ -0,0 +1,403 @@
+
+        pin = randrange(1000, stop=10000)
+        self.pairing.pin(pin)
+        return self.async_show_form(
+            step_id="pair_no_pin",
+            description_placeholders={
+                "protocol": protocol_str(self.protocol),
+                "pin": pin,
+            },
+        )
+
+    async def async_step_service_problem(self, user_input=None):
+        """Inform user that a service will not be added."""
+        if user_input is not None:
+            self.credentials[self.protocol.value] = None
+            return await self.async_begin_pairing()
+
+        return self.async_show_form(
+            step_id="service_problem",
+            description_placeholders={"protocol": protocol_str(self.protocol)},
+        )
+
+    async def _async_get_entry(self, protocol, name, credentials, address):
+        if not is_valid_credentials(credentials):
+            return self.async_abort(reason="invalid_config")
+
+        data = {
+            CONF_PROTOCOL: protocol.value,
+            CONF_NAME: name,
+            CONF_CREDENTIALS: credentials,
+            CONF_ADDRESS: str(address),
+        }
+
+        self._abort_if_unique_id_configured(reload_on_update=False, updates=data)
+
+        return self.async_create_entry(title=name, data=data)
+
+    def _next_protocol_to_pair(self):
+        def _needs_pairing(protocol):
+            if self.atv.get_service(protocol) is None:
+                return False
+            return protocol.value not in self.credentials
+
+        for protocol in PROTOCOL_PRIORITY:
+            if _needs_pairing(protocol):
+                return protocol
+        return None
+
+    def _devices_str(self):
+        return ", ".join(
+            [
+                f"`{atv.name} ({atv.address})`"
+                for atv in self.scan_result
+                if atv.identifier not in self._async_current_ids()
+            ]
+        )
+
+    def _prefill_identifier(self):
+        # Return identifier (address) of one device that has not been paired with
+        for atv in self.scan_result:
+            if atv.identifier not in self._async_current_ids():
+                return str(atv.address)
+        return ""
+
+
+class AppleTVOptionsFlow(config_entries.OptionsFlow):
+    """Handle Apple TV options."""
+
+    def __init__(self, config_entry):
+        """Initialize Apple TV options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Apple TV options."""
+        if user_input is not None:
+            self.options[CONF_START_OFF] = user_input[CONF_START_OFF]
+            return self.async_create_entry(title="", data=self.options)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_START_OFF,
```

Comment:
Why do we need an option for this?

Reply:
This is an old work-around for unexpected behaviors in the Apple TV. If it is asleep and someone connects to it, it will wake up. It CEC is configured, it will also wake up the TV and other devices. So if home Assiatant is restarted for instance, this would turn on the TV. This option makes it possible to work-around that and manually start the integration instead.

Reply:
Do some users expect the TV to start, or can we default to require manual start?

Reply:
No, no one expects that. Manual start by default is really not preferred.

Reply:
Ok.